### PR TITLE
Pull SpriteClient into seagulls engine

### PR DESCRIPTION
**Pull request by ToddiusNoobius**
Making this a PR for now - one thing I haven't figured out yet, is how to make the Sprites enum generic

### CODE REVIEW

File: seagulls-engine/src/seagulls/rendering/_sprite_client.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    IPrinter,
+    Position,
+    Size,
+    Sprite,
+    SpriteComponent,
+    SpriteSheet
+)
+
+
+class Sprites(Enum):
```
Comment (Zeelot) [inline seagulls-engine/src/seagulls/rendering/_sprite_client.py:15]: What if you just use something like this?

```python
class SpritesType(Enum, ABC):
    pass
```
and then in the game:
```python
class Sprites(SpritesType):
  foo: "foo-thing"
```

Comment (ToddiusNoobius) [reply seagulls-engine/src/seagulls/rendering/_sprite_client.py]: Added that - got this error after making the changes i've pushed thus far:

Traceback (most recent call last):
  File "C:\Users\DevAcct\AppData\Local\pypoetry\Cache\virtualenvs\seagulls-rpg-demo-h_FSFkDL-py3.9\Scripts\seagulls", line 5, in <module>
    main()
  File "C:\seagulls-py\seagulls-cli\src\seagulls\seagulls_cli\_entry_point.py", line 9, in main
    app.execute()
  File "C:\seagulls-py\seagulls-cli\src\seagulls\seagulls_cli\_application.py", line 53, in execute
    self._plugin_client.register_plugins(self)
  File "C:\seagulls-py\seagulls-engine\src\seagulls\app\_plugin_client.py", line 45, in register_plugins
    for plugin_ref in self._entrypoint_source.entry_points(self._entrypoint_name):
  File "C:\seagulls-py\seagulls-engine\src\seagulls\app\_plugin_client.py", line 23, in entry_points
    plugin_ref: Type[ISeagullsApplicationPluginRegistrant] = plugin.load()
  File "C:\Users\DevAcct\AppData\Local\pypoetry\Cache\virtualenvs\seagulls-rpg-demo-h_FSFkDL-py3.9\lib\site-packages\importlib_metadata\__init__.py", line 194, in load
    module = import_module(match.group('module'))
  File "C:\Users\DevAcct\.pyenv\pyenv-win\versions\3.9.2\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "C:\seagulls-py\seagulls-rpg-demo\src\seagulls\rpg_demo\__init__.py", line 1, in <module>
    from ._cli_entry_point import RpgDemoCliPluginEntryPoint
  File "C:\seagulls-py\seagulls-rpg-demo\src\seagulls\rpg_demo\_cli_entry_point.py", line 7, in <module>
    from seagulls.rpg_demo._di_container import RpgDemoDiContainer
  File "C:\seagulls-py\seagulls-rpg-demo\src\seagulls\rpg_demo\_di_container.py", line 13, in <module>
    from seagulls.pygame import PygameCameraPrinter, WindowSurface
  File "C:\seagulls-py\seagulls-engine\src\seagulls\pygame\__init__.py", line 9, in <module>
    from ._printer import PygameCameraPrinter, PygamePrinter
  File "C:\seagulls-py\seagulls-engine\src\seagulls\pygame\_printer.py", line 9, in <module>
    from seagulls.rendering import (
  File "C:\seagulls-py\seagulls-engine\src\seagulls\rendering\__init__.py", line 9, in <module>
    from ._sprite_client import SpriteClient, SpritesType, SpriteInfo
  File "C:\seagulls-py\seagulls-engine\src\seagulls\rendering\_sprite_client.py", line 16, in <module>
    class SpritesType(Enum, ABC):
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
C:\seagulls-py\seagulls-rpg-demo>

Comment (Zeelot) [reply seagulls-engine/src/seagulls/rendering/_sprite_client.py]: did some digging and it looks like enums are not like normal classes and we can't just subclass them like everything else :)

but the rules are just that if you extend an enum class, the parent class cannot have any defined values. So it looks like you just have to remove `ABC` from the parent class and things should work!

Comment (ToddiusNoobius) [reply seagulls-engine/src/seagulls/rendering/_sprite_client.py]: That worked!

**Review by Zeelot — APPROVED** [review]

**Pull request merged**